### PR TITLE
Nate/feature/LOOP-1840/carb entry time ux

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1840

Adds a steppable `DateAndDurationTVC`

Screen Shot 2020-09-28 at 09 43 40

### CODE REVIEW

File: LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
+//
+//  DateAndDurationSteppableTableViewCell.swift
+//  LoopKitUI
+//
+//  Copyright © 2018 LoopKit Authors. All rights reserved.
```
Comment (darinkrauss) [inline LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift]: [nit] 2020?

File: LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        let dateFormatter = DateFormatter()
+        dateFormatter.doesRelativeDateFormatting = true
+        dateFormatter.dateStyle = .none
+        dateFormatter.timeStyle = .short
+    
+        return dateFormatter
+    }()
+
+    public override func updateDateLabel() {
+        if dateFormatter.doesRelativeDateFormatting {
```
Comment (darinkrauss) [inline LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift:53]: [nit] Isn't this a constant given 8 lines above?

File: LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            dateFormatter.dateStyle = Calendar.current.isDateInToday(date) ? .none : .short
+        }
+        
+        switch datePicker.datePickerMode {
+        case .countDownTimer:
+            dateLabel.text = durationFormatter.string(from: duration)
+        case .date:
```
Comment (darinkrauss) [inline LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift]: [nit] Consider collapsing these 3 cases into one statement (to reduce duplicated code).

File: LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift (hunk 1 above)
Comment (darinkrauss) [reply LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift]: Ah, I see, this is copied.

**Review by darinkrauss — APPROVED** [review]
A couple of nits, but LGTM! 🚀

File: LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift (hunk 2 above)
Comment (nhamming) [reply LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift:53]: I was just considering that if `doesRelativeDateFormatting` becomes configurable, then the date portion shouldn't be removed for today.

**Pull request merged**